services/api/src/messaging/players.service.ts

```diff
@@ -36,6 +37,33 @@ export const findPlayerById = (
   return player;
 };
 
+export const findPlayerByConnection = (ws: GameWebSocket): Player => {
+  const players = gameRoomService.getPlayers(ws.roomId);
+  const player = players.get(ws);
+  if (!player) {
+    throw new Error('player with this ws connection not found');
+  }
+  return player;
+};
+
+export const checkIfLastToVote = (
+  roomId: string,
+  playerId: string,
+): boolean => {
+  const player = findPlayerById(roomId, playerId);
+  const voters = gameService.getVoters(roomId);
+  const votersWhoMadeMoveCount = gameService.getVoterWhoMadeActionCount(roomId);
+  const playerCount = gameRoomService.getPlayers(roomId).size;
+
+  const isOneVoteMissing = playerCount === votersWhoMadeMoveCount + 1;
+  const hasPlayerMoved = player[1].status !== PlayerStatus.ActionNotTaken;
+
+  if (isOneVoteMissing && voters.length > 1 && !hasPlayerMoved) {
```

Comment:
why `isOneVoteMissing` not enough?

Reply:
I've got rid of `voters.length > 1` condition, but `hasPlayerMoved` is needed, because, we need to know, if the particular player has not moved. It also reflects in method name, which is `checkIfLastToVote`